### ISSUE TITLE
Hiding /special message

### ISSUE THREAD

**Issue by SanderLe** [HasKha/GWToolboxpp#205]
I would like the ability to hide the message "You have a special item available: 'Factions Collector's Edition Present'. Special items can be created only once per account using /special." I don't feel ready to claim it yet, but I would still like to prevent the message from showing up whenever I enter an outpost.

**Comment by HasKha** [comment]
sounds reasonable, but I would need to get the message packet that generates that message, and I don't have an account with that message. 

If you can compile from the source, you would have to uncomment the define here https://github.com/HasKha/GWToolboxpp/blob/master/GWToolbox/GWToolbox/Modules/ChatFilter.cpp#L20, run and check the console. 

Otherwise, I can post a .dll for you to run, then you'll have to report back and I'll post a .dll with the filter for you to verify that its working.

**Comment by SanderLe** [comment]
I'm having issues cloning the GWCA dependency, so I can't compile
[output.txt](https://github.com/HasKha/GWToolboxpp/files/2015150/output.txt)

**Comment by reduf** [comment]
GWCA-public isn't updated yet, but it should be there soon.

**Comment by SanderLe** [comment]
It is a problem, because it causes an abort, preventing other dependencies from being cloned as well.

**Comment by SanderLe** [comment]
I'm still unable to open the workspace after cloning it
[output2.txt](https://github.com/HasKha/GWToolboxpp/files/2015684/output2.txt)
![image](https://user-images.githubusercontent.com/6931955/40218956-be5c177a-5a73-11e8-8737-66a9ebb203ee.png)

**Comment by SanderLe** [comment]
I can't figure out how to add GWCA-Public to the solution, as it lacks a project file. From what I've read at https://docs.microsoft.com/en-gb/visualstudio/ide/how-to-create-and-remove-project-dependencies it seems something needs to be a project in the solution before I can add it as dependency, and I don't know how to make GWCA-Public a project in the solution.

**Comment by HasKha** [comment]
my bad, you don't need to add it to the solution, you only have to adjust the include and library paths to GWToolbox project. Editing my answer above.

**Comment by SanderLe** [comment]
Sorry, I can't get it to work, I think I'll need to take your offer of posting a .dll that I can run

**Comment by SanderLe** [comment]
I'm getting errors related to imgui as well, so it's still not working
![image](https://user-images.githubusercontent.com/6931955/40282866-9df35dd2-5c75-11e8-893b-780f47ebd90b.png)
If I try to comment those out, compile and then try to run the game with it, the game crashes

**Comment by HasKha** [comment]
about imgui, you must be on a different, probably more recent, version/commit of imgui. Make sure you're on v1.53. Submodules should automatically set to that, maybe `git submodule update` (and ignore the errors related to gwca repo).

**Comment by MaddoW** [comment]
I just tried to build and run it with latest versions of imgui and public gwca, it works for me. Did you replaced the .dll from ../Release in your C:/USERs/USER/AppData/Local/GWToolboxpp folder?

**Comment by SanderLe** [comment]
Well, I got it to run, but I don't know how to see the chat packets. If I try to run it in Visual Studio, it says Toolbox.dll is not a valid Win32 application. If I try to attach it to Gw.exe in Visual Studio, nothing noticeable happens to the game. And if I put it in AppData\Local\GWToolboxpp, I don't see a noticeable difference with a regular GWToolbox.dll, I don't see anywhere to read the chat packets.

**Comment by SanderLe** [comment]
[log.txt](https://github.com/HasKha/GWToolboxpp/files/2023318/log.txt)
I used /bonusitems first, the first set of chat packets comes from that, the last one is related to the /special reward
Then I map travelled to trigger the /special message normally

**Comment by SanderLe** [comment]
TLDR: The option to block /special messages can be implemented by blocking messages starting with 0x8101 0x39CD 0x10A and with P082: id 5. Probably... I'm no expert.

**Comment by HasKha** [comment]
good job getting it to work, and thanks again. 

your tldr is most likely correct, I'll just double check when I find the time to do this.

**Comment by HasKha** [comment]
adding this, will be in next release. Binding the toggle to the 'nine-rings' filter, since I don't want to make an extra option that is only needed by a handful of users, and I don't want to force hide it, in case anything is off and it hides more than what we want.

Since you have a local copy, you can test the change yourself as well. In order to do so, open ChatFilter.cpp again, go to line 353 and add:
```
        case 0x39CD: // you have a special item available: <special item reward>
            return ninerings;
```
as shown here: https://github.com/HasKha/GWToolboxpp/commit/766c305530ee5a73b5d58e852dd25bceb1f81dec
